ui/pages/bridge/utils/useBridgeData.ts

```diff
@@ -0,0 +1,62 @@
+import { useSelector } from 'react-redux';
+import { Hex } from '@metamask/utils';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useHistory } from 'react-router-dom';
+import { selectBridgeHistoryForAccount } from '../../../ducks/bridge-status/selectors';
+import { CROSS_CHAIN_SWAP_TX_DETAILS_ROUTE } from '../../../helpers/constants/routes';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import useBridgeChainInfo from './useBridgeChainInfo';
+
+export type UseBridgeDataProps = {
+  transactionGroup: {
+    hasCancelled: boolean;
+    hasRetried: boolean;
+    initialTransaction: TransactionMeta;
+    nonce: Hex;
+    primaryTransaction: TransactionMeta;
+    transactions: TransactionMeta[];
+  };
+};
+
+export default function useBridgeData({
+  transactionGroup,
+}: UseBridgeDataProps) {
+  const t = useI18nContext();
+  const history = useHistory();
+  const bridgeHistory = useSelector(selectBridgeHistoryForAccount);
+
+  const srcTxHash = transactionGroup.initialTransaction.hash;
+
+  // If this tx is a bridge tx, it will have a bridgeHistoryItem
+  const bridgeHistoryItem = srcTxHash ? bridgeHistory[srcTxHash] : undefined;
+
+  const { destNetworkConfiguration } = useBridgeChainInfo({
+    bridgeHistoryItem,
+  });
+
+  const destChainName = destNetworkConfiguration?.name;
```

Comment:
Can you use a constant to look this up? If the user doesn't have the dest network added this will be undefined

Reply:
Done in [49ee271](https://github.com/MetaMask/metamask-extension/pull/27740/commits/49ee271ed3c72d8cd80e9c3dcefd5d6a09ae2cb5)